analysis/warlockaffliction/src/modules/covenants/DecimatingBolt.tsx

```diff
@@ -0,0 +1,37 @@
+import SPELLS from 'common/SPELLS';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+class DecimatingBolt extends Analyzer {
+  _decimatingBoltDamage: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.NECROLORD.id);
+
+    this.addEventListener(
```

Comment:
It's not *that* big of a deal, but in all cases where the module might *not* be active, check for it and return early. The way you do it, the event listener is added even if th player doesn't have Necrolord covenant.
```ts
this.active = ...;
if (!this.active) {
  return;
}

this.addEventListener(...);
```